src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -1144,7 +1144,16 @@ object LazyList extends SeqFactory[LazyList] {
     /** Construct a LazyList consisting of the concatenation of the given LazyList and
       *  another LazyList.
       */
-    def #:::[B >: A](prefix: LazyList[B]): LazyList[B] = prefix lazyAppendedAll l()
+    def #:::[B >: A](prefix: => LazyList[B]): LazyList[B] = newLL(prefix.state) lazyAppendedAll l()
+
+    /** @deprecated Use the #::: method with a by-name parameter instead.
+      *             This can be accomplished by simply recompiling.
+      */
+    @deprecated(
+      "Use the #::: method with a by-name parameter instead; see docs for more info",
+      since = "2.13.15"
+    )
+    def #:::[B >: A]()(prefix: LazyList[B]): LazyList[B] = prefix lazyAppendedAll l()
```

Comment:
this is a trick I learned from @‍armanbilge (deliberate fake tag, I don't want to ping) to maintain bincompat but also prevent source collision

Reply:
You can even declare it as `protected[immutable]` still that it doesn't appear in the public API at all.